clinica/utils/io.py

```diff
@@ -130,11 +130,23 @@ def reorder_bids_or_caps_files(input_ids, bids_or_caps_files):
 
 
 def check_bids_folder(bids_directory):
-    import os
+    """
+    check_bids_folder function checks the following items :
+        - bids_directory is a string
+        - the provided path exists and is a directory
+        - provided path is not a CAPS folder (BIDS and CAPS could be swapped by user). We simply check that there is
+          not a folder called 'subjects' in the provided path (that exists in CAPS hierarchy)
+        - provided folder is not empty
+        - provided folder must contains at least one directory whose name starts with 'sub-'
+    """
+    from os.path import isdir, join
+    from os import listdir
     from colorama import Fore
     from clinica.utils.exceptions import ClinicaBIDSError
 
-    if not os.path.isdir(bids_directory):
+    assert isinstance(bids_directory, str), 'Argument you provided to check_bids_folder() is not a string.'
+
+    if not isdir(bids_directory):
```

Comment:
The code to raise Exception is not harmonized in the sense that we can have:
- `ClinicaBIDSError("\n%s[Error] Clinica message.%s\n" % (Fore.BLUE, Fore.RESET))`
- `ClinicaBIDSError(Fore.RED + '\n[Error] Clinica message.' + Fore.RESET)`

I let you chose for `"` vs `'` and `%s %(var)` vs `+ var +` but it needs to be the same for `check_{bids|caps}_folder` functions.